### PR TITLE
Change locations for Resource files

### PR DESCRIPTION
**Pull request by shresthagrawal**
* [x] marshal and un marshal file ref data
  * [x] make required changes in xml file ref
  * [x] make parser save the modified xmljs object back
  * [x] recursively find all the data ref 
  * [x] change location relatively. 

Work still in Progress!!

### CODE REVIEW

**Comment by shresthagrawal** [comment]
@falafeljan can you review the code. I am not sure why I the test fails in the test environment. Let me know if you find something or else I will just remove that test.

File: test/fileref-test.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+0000FFFFFFFF000000004149464600000000FFFFFFFF00000A206375\
+00000000000000000000000000074465736B746F700000";
+
+const footer = "1300012F00001500020015FFFF0000";
+
+// TODO: Use a relative location as this location would not exist in every system.
+const location = "Users/shresthagrawal/Desktop/0001 20-Audio.aif";
+const newLocation = "/Users/shresthagrawal/Desktop/a/Desktop/0001 20-Audio.aif";
+const systemName = "Macintosh HD";
+const hex = "48656C6C6F20576F726C6421313233344023";
```
[the remark this answers is not in the document]
Comment (shresthagrawal) [reply test/fileref-test.ts]: Hey how do you expect to generate the stream blob for the test env?

File: test/parser-test.ts [hunk 2]
```diff
@@ -14,11 +15,27 @@ const expect = chai.expect;
...
+  'Users/shresthagrawal/Desktop/work/GitMusic/ableton-parser/test/res/resource-test/a/d/audio.aif',
+  'Users/ama/Downloads/Reverb Default.adv',
+  'Users/mak/Library/Application Support/Ableton/Live 10 Core Library/Devices/Audio Effects/Simple Delay/Dotted Eighth Note.adv'
+];
+
+const modifiedResource = [
+  'Users/shresthagrawal/Desktop/work/GitMusic/ableton-parser/test/res/resource-test/b/d/audio.aif',
+  'Users/ama/Downloads/Reverb Default.adv',
+  'Users/mak/Library/Application Support/Ableton/Live 10 Core Library/Devices/Audio Effects/Simple Delay/Dotted Eighth Note.adv'
+]
```
[the remark this answers is not in the document]
Comment (shresthagrawal) [reply test/parser-test.ts]: same as above

File: test/parser-test.ts [hunk 3]
```diff
@@ -49,4 +66,26 @@ describe('Parser', function() {
...
+        });
+        it('Get the list of resourcefiles when als project file is given', function() {
+            this.parserA.getResourceLocations().should.eql(resources);
+        });
+        it('Change location of all resourcefiles when als project file is given', async function() {
+            let newlocation = './test/res/resource-test/b/d/';
+            this.parserA.changeResourceLocations(newlocation);
+            let modifiedProject = await parseFile(path.join(tmpDir2, projectA));
+            modifiedProject.getResourceLocations().should.eql(modifiedResource);
+        });
```
[the remark this answers is not in the document]
Comment (shresthagrawal) [reply test/parser-test.ts]: sounds like a good idea!

**Comment by falafeljan** [comment]
@shresthagrawal another idea would be using an in-memory filesystem, so we don't have to deal with relative paths and can use any path in theory. memfs is suited for that: https://github.com/streamich/memfs.

**Comment by shresthagrawal** [comment]
@falafeljan Hey its really a pain to generate blobs and dataset for any change in the path, I spent a lot of time on this today. I thinks the parser works quite decently. There isn't a better way than generating the blobs automatically. I would prefer not spending time on this at the moment.

**Comment by falafeljan** [comment]
alright, it was just a remark for further work. as i said, it's terrific work and i'm sure it's ready to merge!

**Pull request merged**